src/NBitcoin/Protocol/Node.cs

```diff
@@ -486,8 +482,7 @@ internal Node(NetworkAddress peer, Network network, NodeConnectionParameters par
                     Utils.SafeCloseSocket(socket);
                     NodeServerTrace.Information("Connection to node cancelled");
                     this.State = NodeState.Offline;
-                    if (addrman != null)
```

Comment:
Make sure this is also replace din the addnode RPC method

Reply:
As discussed this has been checked 👍